proposals/lambda-method-group-defaults.md

```diff
@@ -0,0 +1,336 @@
+
+In the case of method groups, we will continue to allow the implicit conversion to delegate with no warning, even if 
+there are no defaults in the target delegate type in order to keep our existing behavior. For more information on method group conversions,
+see [§10.8](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/conversions.md#108-method-group-conversions).
+
+Note that the core distinction here is that with a method group, the underlying method can still
+be called by name in the standard way whereas if a lambda is assigned to a delegate, it can only
+be called through the delegate. 
+
+```csharp
+void Fun(int i = 4) { ... }
+
+delegate void D1(int i);
+D1 d = (int i = 1) => 42; // Allowed WITH warning, since the lambda is now only callable through the delegate
+                          // and the default parameter has effectively been thrown away.
+
+D1 d = Fun; // Allowed WITHOUT warning, since Fun and its default parameter are still accessible
+            // without using the delegate.
+```
+
+### IL/Runtime Behavior
+
+The default parameter values will be emitted to metadata. The IL for this feature will be very similar in nature to the IL emitted for lambdas with `ref` and `out` parameters. A class which
+inherits from `System.Delegate` or similar will be generated, and the `Invoke` method will include `.param` directives to set default parameters -- 
+just as would be the case for a standard named delegate with default parameters.
+
+These delegate types are able to inspected at runtime, as normal.
+In code, users can introspect the `DefaultValue` in the `ParameterInfo` associated with the lambda or method group by using the associated `MethodInfo`.
+
+```csharp
+var addWithDefault = (int addTo = 2) => addTo + 1;
+void addWithDefaultMethod(int addTo = 2) {
+  return addTo + 1;
+}
+
+addWithDefault.Method.GetParameters()[0].DefaultValue; // 2
+
+var add1 = addWithDefaultMethod;
+add1.Method.GetParameters()[0].DefaultValue; // 2
+```
+
+## Open Questions
+
+**Open question:** how does this interact with the existing `DefaultParameterValue` attribute?
+
+**Proposed answer:** For parity, permit the `DefaultParameterValue` attribute on lambdas and ensure that the delegate generation behavior matches for default parameter values supported via the syntax.
+
+```csharp
+var a = (int i = 13) => 1;
+// same as
+var b = ([DefaultParameterValue(13)] int i) => 1;
+b = a; // Allowed
+```
+
+**Open question:** First, note that this is outside the scope of the current proposal but it might be worth discussing in the future.
+Do we want to support defaults with implicitly typed lambda parameters? I.e., 
+
+```csharp
+delegate void M1(int i = 3);
+M1 m = (x = 3) => x + x; // Ok
+
+delegate void M2(long i = 2);
+M2 m = (x = 3.0) => ...; //Error: cannot convert implicitly from long to double
+```
+This inference leads to some tricky conversion issues which would require more discussion.
```

Comment:
It's not just inference that is a problem, it would increase the look ahead for our parser. Consider the state of the parser when it's looking at the expression that starts with a `(`. It has to decide if that is a lambda expression or not. Today the parser can bail out the moment it gets to `(x =` as that disqualifies it from being a lambda. If we allowed defaults on implicit then it has to parse all the way through the `=>`.

Reply:
Ah thanks for pointing this out. I will add a note about that as well